### PR TITLE
Add a Backlog API endpoint

### LINKED ISSUE #7393: Provide an api to gather the backlog
There is an API to list the staged_requests and one to list the excluded_requests

I'm missing an API to list the backlog. Right now the client has to get the requests for by_group=managers and substract the excluded_requests manually.

### PR DESCRIPTION
**Pull request by danidoni**
This PR adds a new API endpoint under `/staging_workflow/:project_name/backlog`

And the output is:
```                                                                                                           
<backlog>
  <request id="6" type="delete" creator="Iggy" state="review" package="doom"/>
</backlog>
```

Fixes #7393

Co-authored-by: David Kang <dkang@suse.com>

### CODE REVIEW

File: src/api/app/controllers/staging/backlog_controller.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  before_action :set_staging_workflow
+
+  def index
+    @backlog = @staging_workflow.unassigned_requests
+  end
+
+  private
+
+  def set_project
+    @project = Project.get_by_name(params[:staging_workflow_project])
```
[the remark this answers is not in the document]
Comment (DavidKang) [reply src/api/app/controllers/staging/backlog_controller.rb]: `get_by_name` throw an exception. We have overwritten this method.

**Comment by DavidKang** [comment]
@vpereira,

>  I see in many Staging related controllers the same set_project method. Wouldn't be better to write it just once?

Yep, it would be better, but IMO, it's out of the scope. We can create different PR for that.

**Comment by adrianschroeter** [comment]
On Freitag, 11. Oktober 2019, 08:09:00 CEST David Kang wrote:
> DavidKang commented on this pull request.
> 
> > @@ -0,0 +1,22 @@
> 
> +class Staging::BacklogController < ApplicationController
> +  before_action :require_login, except: [:index]
> +  before_action :set_project
> +  before_action :set_staging_workflow
> +
> +  def index
> +    @backlog = @staging_workflow.unassigned_requests
> +  end
> +
> +  private
> +
> +  def set_project
> +    @project = Project.get_by_name(params[:staging_workflow_project])
> 
> `get_by_name` throw an exception. We have overwritten this method.

argh, this is security functionality, never ever overwrite this ....

-- 

Adrian Schroeter <adrian@suse.de>
Build Infrastructure Project Manager

SUSE Software Solutions Germany GmbH,  Maxfeldstr. 5, 90409 Nuernberg, Germany
(HRB 247165, AG München), Geschäftsführer: Felix Imendörffer

**Comment by DavidKang** [comment]
> argh, this is security functionality, never ever overwrite this ...

@adrianschroeter we didn't touch this method, I was thinking about other methods.

**Pull request merged**